File: lingting-core/src/main/java/live/lingting/framework/api/PaginationResult.java
package live.lingting.framework.api;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.Collections;
import java.util.List;

/**
 * @author lingting 2024-02-02 17:53
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
public class PaginationResult<T> {

	private Long total = 0L;

	private List<T> records = Collections.emptyList();

}

File: lingting-core/src/main/kotlin/live/lingting/framework/api/PaginationResult.kt
package live.lingting.framework.api

import java.util.function.Consumer
import java.util.function.Function

/**
 * @author lingting 2024-02-02 17:53
 */
data class PaginationResult<T> @JvmOverloads constructor(
    var total: Long = 0,
    var records: List<T> = emptyList(),
) {

    constructor(records: List<T>) : this(records.size.toLong(), records)

    fun <V : Any> map(function: Function<T, V>): PaginationResult<V> {
        return PaginationResult(total, records.map { function.apply(it) })
    }

    fun forEach(consumer: Consumer<T>) {
        records.forEach(consumer)
    }

}
